doc/architectural/compilation-and-development.md

```diff
@@ -3,43 +3,106 @@
-input files and one or more `.desc` files that describe what command
-to run, what output is expected and so on. There is a Perl script,
-`test.pl` that is used to invoke the tests as:
+
+\subsection compilation-and-development-subsection-personal-configuration Personal configuration
+
+\subsubsection compilation-and-development-subsubsection-config-inc config.inc
+
+Two files,
+[config.inc](https://github.com/diffblue/cbmc/blob/develop/src/config.inc) and
+[common](https://github.com/diffblue/cbmc/blob/develop/src/common), are
+included in every makefile. 
+[config.inc](https://github.com/diffblue/cbmc/blob/develop/src/config.inc)
+contains configuration options
+relating to compilation so that they can be conveniently edited in one place.
+[common](https://github.com/diffblue/cbmc/blob/develop/src/common)
+contains commands that are needed in every makefile but which the
+developer is not expected to edit. (There is also
+[another config.inc](https://github.com/diffblue/cbmc/blob/develop/jbmc/src/config.inc)
+which is also included in every makefile in the `jbmc` folder.) 
+ 
+
+\subsubsection compilation-and-development-subsubsection-macro-debug Macro DEBUG
+
+If the macro `DEBUG` is defined during compilation of CBMC (for example by
+using a compiler flag) then extra debug code will be included. This includes
+print statements and code checking that data structures are as expected.
+
+
+\section compilation-and-development-section-running-tests Running tests
+
+\subsection compilation-and-development-subsection-regression-tests Regression tests
+
+The regression tests are contained in the `regression/` folder. Inside
+`regression/` there is a directory for each of the tools/modules. Each of
+these contains multiple test directories, with names describing
+what they test. When there are multiple tests in a test directory then
+they should all be testing the same thing. Each test directory contains
+input files and one or more test description files,
+which have the ending `.desc`. The test description files describe what command
+to run, what output is expected and so on. The test framework is an in-house
+Perl script,
+[test.pl](https://github.com/diffblue/cbmc/blob/develop/regression/test.pl).
+To run all the tests in a directory corresponding to a tool or module:
 
     ../test.pl -c PATH_TO_CBMC
 
-The `–help` option gives instructions for use and the
-format of the description files.
+The `--help` option to `test.pl` gives instructions for use and outlines the
+format of the test description files.
+
+If you have compiled CBMC using CMake then another way to run the same tests
+is through `CTest`. From the build directory, the command `ctest -V -L CORE`
+will run all of the tests that are run in CI. `-V` makes it print out more
+useful output and `-L CORE` makes it only run tests that have been tagged
+`CORE`. `-R regular_expression` can be used to limit which tests are run, and
```

Comment:
Might be worth mentioning this _can't_ be used to for example run a specific `.desc` file, only one of the top level folders.

Reply:
Is there actually a way to run individual tests in CMake? Not knowing how to work with the tests is one of the reasons I've largely refrained from adopting CMake.

Reply:
I believe that what we add to CTest is one command per module/tool folder, so CTest doesn't know how to run just one folder within one of those.